### PR TITLE
remove actions and jenkins; this plugin needs revamp

### PR DESCRIPTION
**Pull request by erik-east**
this repo doesn't work locally, while I was able to get some of the make commands to work with minor updates overall this plugin needs a complete overhaul.

I'm able to run `make docker` and then `docker run --name containerName imageName` and get an exec command error due to it being build on an amd64 architecture not a x86 one.  however the `docker-compose up -d` fails entirely.  

regardless we've decided to archive the repos that need an overhaul.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        run: |
+          echo "GOLANG_VERSION=$(go version | awk '{ print $3 }')" >> $GITHUB_ENV
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
```
Comment (sgardnervio) [inline .github/workflows/release.yml:27]: its strange seeing a hyphen on a blank line

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: Set GOLANG_VERSION
```
Comment (sgardnervio) [inline .github/workflows/release.yml:17]: not sure that this is needed as I don't see the GOLANG_VERSION being used.. is it merely for console output?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  snapshot-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: Set GOLANG_VERSION
```
Comment (sgardnervio) [inline .github/workflows/build.yml:38]: I don't see the GOLANG_VERSION being used anywhere .. why set it?

**Review by schoterson — COMMENTED** [review]
looks like the actions work :)

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - run: "go vet ./..."
+      - name: Lint
+        uses: dominikh/staticcheck-action@v1.2.0
+        with:
+          version: "2022.1.1"
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
```
Comment (wwitzel3) [inline .github/workflows/build.yml:25]: `uses: actions/setup-go@v4`

File: .github/workflows/build.yml (hunk 3 above)
Comment (GuessWhoSamFoo) [reply .github/workflows/build.yml:38]: It is used as a build flag via ldflags (https://github.com/vapor-ware/synse-snmp-plugin/blob/master/Makefile#L19) and goreleaser (https://github.com/vapor-ware/synse-snmp-plugin/blob/master/.goreleaser.yml#L14) using the same mechanism

File: .github/workflows/release.yml (hunk 1 above)
Comment (GuessWhoSamFoo) [reply .github/workflows/release.yml:27]: This is valid yaml although we should enforce with tooling if preferred to see a specific way

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - run: "go vet ./..."
+      - name: Lint
+        uses: dominikh/staticcheck-action@v1.2.0
```
Comment (GuessWhoSamFoo) [inline .github/workflows/build.yml:18]: we're on 1.3 which gets rids of the `set-output` deprecation warnings

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: Set GOLANG_VERSION
+        run: |
+          echo "GOLANG_VERSION=$(go version | awk '{ print $3 }')" >> $GITHUB_ENV
+      - name: Snapshot Build
+        uses: goreleaser/goreleaser-action@v3
```
Comment (GuessWhoSamFoo) [inline .github/workflows/build.yml:42]: We are also on `goreleaser-action@v4` now, remove `--debug` flag (https://goreleaser.com/deprecations/#-debug)

**Comment by GuessWhoSamFoo** [comment]
New linter is more strict than the old one on jenkins. My opinion is that they're reasonable, however feel free to tune it differently

File: .github/workflows/build.yml (hunk 4 above)
Comment (GuessWhoSamFoo) [reply .github/workflows/build.yml:25]: I've be setting to the one used in go mod - we could use a matrix if inclined

**Comment by sgardnervio** [comment]
after discussion with kyler, dillon, steve, erk, nidesh its decided that these plugins which can't be worked on locally due to being out of date should not have GHA or jenkins until they are re-written to follow an updated plugin sdk which is TBD

**Pull request merged**